rfd/0086/config.md

```diff
@@ -2,174 +2,236 @@
+    {
+      name: "appB",
+      port: 8000
+    }
+  ],
+  health: [
+    {
+      name: "checkB",
+      service: "nginx"
+      exec: "curl -s --fail localhost/otherhealth"
+    }
+  ]
+}
 ```
 
 These will be merged as follows:
 
-```
-consul:
-  host: consul.svc.triton.zone
-
-services:
-  nginx:
-    port: 80
-
-  app-A:
-    port: 9000
-
-  app-B:
-    port: 8000
-
-health:
-  nginx:
-    check-A:
-      command: curl -s --fail localhost/health
-    check-B:
-      command: curl -s --fail localhost/otherhealth
-
+```json5
+{
+  consul: {
+    host: "consul.svc.triton.zone:8500"
+  },
+  jobs: [
+    {
+      name: "nginx",
+      port: 80
+    },
+    {
+      name: "appA",
+      port: 9000
+    },
+    {
+      name: "appB",
+      port: 8000
+    }
+  ],
+  health: [
+    {
+      name: "checkA",
+      service: "nginx"
+      exec: "curl -s --fail localhost/health"
+    },
+    {
+      name: "checkB",
+      service: "nginx"
+      exec: "curl -s --fail localhost/otherhealth"
+    }
+  ]
+}
 ```
 
 
 The full example configuration for ContainerPilot found in the existing docs would look like the following:
 
 
-```
-consul:
-  host: localhost:8500
-
-logging:
-  level: INFO
-  format: default
-  output: stdout
-
-service:
-  app:
-    port: 80
-    heartbeat: 5
-    tll: 10
-    tags:
-    - app
-    - prod
-    interfaces:
-    - eth0
-    - eth1[1]
-    - 192.168.0.0/16
-    - 2001:db8::/64
-    - eth2:inet
-    - eth2:inet6
-    - inet
-    - inet6
-    - static:192.168.1.100"
-
-    stopTimeout: 5
-    preStop: /usr/local/bin/preStop-script.sh
-    postStop: /usr/local/bin/postStop-script.sh
-
-    depends:
-      setup:
-        wait: success
-      nginx:
-        onChange: /usr/local/bin/reload-nginx.sh
-        poll: 30
-        timeout: "30s"
-      consul-agent:
-        wait: healthy
-      app:
-        onChange: /usr/local/bin/reload-app.sh
-        poll: 10
-        timeout: "10s"
-
-  setup:
-    command: /usr/local/bin/preStart-script.sh {{.ENV_VAR_NAME}}
-    advertise: false
-    restart: never
-
-  consul-agent:
-    port: 8500
-    command: consul -agent -join consul
-    advertise: false
-    restart: always
-
-  consul-template:
-    command: >
-      consul-template -consul consul
-          -template /tmp/template.ctmpl:/tmp/result
-    advertise: false
-    restart: always
-
-  task1:
-    command: /usr/local/bin/tash.sh arg1
-    frequency: 1500ms
-    timeout: 100ms
-    advertise: false
-
-
-health:
-  nginx:
-    check-A:
-      command: >
-        /usr/bin/curl --fail -s -o /dev/null http://localhost/app
+```json5
+{
+  consul: {
+    host: "localhost:8500"
+  },
+  logging: {
+    level: "INFO",
+    format: "default",
+    output: "stdout"
+  },
+  jobs: [
+    {
+      name: "app",
+      // this is upstart-like syntax indicating we want to start this
+      // service when the "setup" service has exited with success but
+      // give up after 60 sec
+      when: "setup exitSuccess timeout 60s",
```

Comment:
Where should I look to see the syntax for the events described? Specifically, I'm looking for more background on the choice of this syntax:

`when: "setup exitSuccess timeout 60s"`

Over something like:

```json
when: {
    job: "setup",
    event: "exitSuccess",
    timeout: "60s",
}
```

I'm not sure I have an opinion, I was just trying to better understand the the reasoning to use different syntax for defining those components, rather than the JSON5 we're using elsewhere.